physics/test/ThermoIce0_test.cpp

```diff
@@ -0,0 +1,142 @@
+/*!
+ * @file ThermoIce0Temperature_test.cpp
+ *
+ * @date Apr 29, 2022
+ * @author Tim Spain <timothy.spain@nersc.no>
+ */
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+#include <sstream>
+
+#include "include/ThermoIce0.hpp"
+#include "include/IFluxCalculation.hpp"
+
+#include "include/Configurator.hpp"
+#include "include/ConfiguredModule.hpp"
+#include "include/IFreezingPoint.hpp"
+#include "include/IFreezingPointModule.hpp"
+#include "include/ModelArray.hpp"
+#include "include/ModelArrayRef.hpp"
+#include "include/ModelComponent.hpp"
+#include "include/Time.hpp"
+
+namespace Nextsim {
+
+TEST_SUITE_BEGIN("ThermoIce0");
+TEST_CASE("Threshold ice")
+{
+    ModelArray::setDimensions(ModelArray::Type::H, { 1, 1 });
+    ModelArray::setDimensions(ModelArray::Type::Z, { 1, 1, 1 });
+
+    class IceTemperatureData : public ModelComponent {
+    public:
+        IceTemperatureData()
+        {
+            registerSharedArray(SharedArray::H_ICE, &hice);
+            registerSharedArray(SharedArray::C_ICE, &cice);
+            registerSharedArray(SharedArray::H_SNOW, &hsnow);
+            registerProtectedArray(ProtectedArray::HTRUE_ICE, &hice0);
+            registerProtectedArray(ProtectedArray::SST, &sst);
+            registerProtectedArray(ProtectedArray::SSS, &sss);
+            registerProtectedArray(ProtectedArray::TF, &tf);
+            registerProtectedArray(ProtectedArray::SNOW, &snow);
+            registerProtectedArray(ProtectedArray::ML_BULK_CP, &mlbhc);
+            registerProtectedArray(ProtectedArray::T_ICE, &tice0);
+            registerSharedArray(SharedArray::Q_IO, &qio);
+        }
+        std::string getName() const override { return "IceTemperatureData"; }
+
+        void setData(const ModelState::DataMap&) override
+        {
+            cice[0] = 0.99;
+            hice[0] = 0.001 / cice[0]; // Here we are using the true thicknesses
+            hice0[0] = hice[0];
+            hsnow[0] = 0.;
+            sss[0] = 32.;
+            sst[0] = Module::getImplementation<IFreezingPoint>()(sss[0]);
+            snow[0] = 0.;
+            tf[0] = Module::getImplementation<IFreezingPoint>()(sss[0]);
+            mlbhc[0] = 4.29151e7;
+            tice0[0] = -9.;
+            qio[0] = 0.;
+        }
+
+        HField hice;
+        HField hice0;
+        HField cice;
+        HField hsnow;
+        HField sst;
+        HField sss;
+        HField tf;
+        HField snow;
+        HField mlbhc; // Mixed layer bulk heat capacity
```

Comment:
1. This is a copy and paste
2. The naming convention clearly isn't sufficient, please re-name